src/eal.cpp

```diff
@@ -106,59 +105,45 @@ namespace eal {
     return coordinate;
   }
 
-  // Interpolation helper functions
-  double linearInterpolate(vector<double> points, vector<double> times, double time) {
-    size_t numPoints = points.size();
-    if (numPoints < 2) {
-      throw invalid_argument("At least two points must be input to interpolate over.");
-    }
-    if (points.size() != times.size()) {
-      throw invalid_argument("Invalid interpolation data, must have the same number of points as times.");
-    }
-    if (time < times.front() || time > times.back()) {
-      throw invalid_argument("Invalid interpolation time, outside of input times.");
-    }
-
-    // GSL setup
-    gsl_interp *interpolator = gsl_interp_alloc(gsl_interp_linear, numPoints);
-    gsl_interp_init(interpolator, &times[0], &points[0], numPoints);
-    gsl_interp_accel *acc = gsl_interp_accel_alloc();
 
-    // GSL evaluate
-    double result = gsl_interp_eval(interpolator, &times[0], &points[0], time, acc);
-
-    // GSL clean up
-    gsl_interp_free(interpolator);
-    gsl_interp_accel_free(acc);
-
-    return result;
-  }
-
-  double splineInterpolate(vector<double> points, vector<double> times, double time) {
-    size_t numPoints = points.size();
-    if (numPoints < 2) {
-      throw invalid_argument("Invalid interpolation data, at least two points are required to interpolate.");
-    }
-    if (points.size() != times.size()) {
-      throw invalid_argument("Invalid interpolation data, must have the same number of points as times.");
-    }
-    if (time < times.front() || time > times.back()) {
-      throw invalid_argument("Invalid interpolation time, outside of input times.");
-    }
-
-    // GSL setup
-    gsl_interp *interpolator = gsl_interp_alloc(gsl_interp_cspline, numPoints);
-    gsl_interp_init(interpolator, &times[0], &points[0], numPoints);
-    gsl_interp_accel *acc = gsl_interp_accel_alloc();
+ double interpolate(vector<double> points, vector<double> times, double time, const gsl_interp_type* interp, derivation d) {
```

Comment:
Smashed the different interpolation functions into one since they all have the same code with the exception of the enum val you pipe into GSL.

Reply:
:+1: